packages/taxdat/R/config_helpers.R

```diff
@@ -14,184 +14,63 @@ automate_mapping_config <- function(p, covariate_names, countries_names, countri
-    "beta_sigma_scale: ", p$beta_sigma_scale, "\n",
-    "sigma_eta_scale: ", p$sigma_eta_scale, "\n",
-    "stan:\n",
-    "  ncores: ", p$ncores, "\n",
-    "  model: '", p$model, ".stan'\n",
-    "  niter: ", p$niter, "\n",
-    "  recompile: ", p$recompile, "\n"
-  ))
-}
+  unspecified_parameter_check <- function(param = NULL, null_check = is.null(param)){
+    if(null_check){return(TRUE)}
 
-#' @title Automate generation of config files -- just for the Aug 2022 runs 
-#' @name auto_config_Aug2022
-#' @description Automate generation of model configuration files from a dataframe of parameters. Generates one config per row in the p dataframe.
-#' @param p dataframe with config parameters, one config per row
-#' @param covariate_names character vector with names of covariates to include in the model
-#' @param countries_names character vector with country codes to include in the model
-#' @param countries_ids character vector with country location ids to include in the model
-#' @return string with yaml config file 
-#' @export
-auto_config_Aug2022 <- function(p, covariate_names, countries_names, countries_ids){
-  
-  map_name <- paste0(paste(countries_names, collapse="_"), "_", lubridate::year(as.Date(p$start_time)), "_", lubridate::year(as.Date(p$end_time)), "_", p$res_space, "km")
+    if(is.null(param) | identical(param, character(0)) | (stringr::str_count(param, " ") == nchar(param))){
+      return(TRUE)
+    }else{
+      return(FALSE)
+    }
+  }
 
   cat(paste0(
     "name: '", map_name, "'\n",
     "countries: ['", paste(countries_ids, collapse="','"), "']\n",
     "countries_name: ['", paste(countries_names, collapse="','"), "']\n",
-    "OCs: ", "\n",
     "aoi: '", p$aoi, "'\n",
     "res_space: ", p$res_space, "\n",
     "res_time: '", p$res_time, "'\n",
     "smoothing_period: ", p$smoothing_period, "\n",
     "case_definition: '", p$case_definition, "'\n",
     "start_time: '", as.Date(p$start_time), "'\n",
     "end_time: '", as.Date(p$end_time), "'\n",
-    { if(is.null(covariate_names) | identical(covariate_names, character(0)) | identical(covariate_names, "")) "covariate_choices:\n" 
-      else paste0("covariate_choices: ['", paste(covariate_names, collapse="','"), "']\n") }, 
     "data_source: '", p$data_source, "'\n",
     "ovrt_metadata_table: no\n",
-    "ingest_covariates: ", p$ingest_covariates, "\n",
-    "ingest_new_covariates: no\n",
-    "warmup: ", p$warmup, "\n",
-    "covar_warmup: ", p$covar_warmup, "\n",
-    "censoring: ", p$censoring, "\n",
-    "tfrac_thresh: ", p$tfrac_thresh, "\n",
-    "aggregate: ", p$aggregate, "\n",
-    "use_weights: ", p$use_weights, "\n",
-    "use_pop_weight: ", p$use_pop_weight, "\n",
-    "time_effect: ", p$time_effect, "\n",
-    "time_effect_autocorr: ", p$time_effect_autocorr, "\n",
-    "beta_sigma_scale: ", p$beta_sigma_scale, "\n",
-    "sigma_eta_scale: ", p$sigma_eta_scale, "\n",
+    { if(unspecified_parameter_check(p$OCs)) "" 
```

Comment:
@KaiyueZou It's not a good idea to hard-code the default parameters in this function when we have the default parameters specified elsewhere in the pipeline. We should instead be pulling the parameters from the places it is already specified to prevent errors.